lib/ansible/plugins/action/__init__.py

```diff
@@ -260,7 +273,7 @@ def _is_pipelining_enabled(self, module_style, wrap_async=False):
             module_style == "new",                     # old style modules do not support pipelining
             not C.DEFAULT_KEEP_REMOTE_FILES,           # user wants remote files
             not wrap_async or self._connection.always_pipeline_modules,  # async does not normally support pipelining unless it does (eg winrm)
-            self._play_context.become_method != 'su',  # su does not work with pipelining,
+            (self._connection.become.name if self._connection.become else '') != 'su',  # su does not work with pipelining,
```

Comment:
I was going to recommend something else here, as it wasn't clear to me when originally reading it, but I'm not sure there is a better way.

In the future, we should probably have some sort of noop become plugin that is the default, to make things like this easier.